pkg/distributor/distributor_test.go

```diff
@@ -5184,3 +5186,60 @@ func TestStartFinishRequest(t *testing.T) {
 		})
 	}
 }
+
+func TestSendMessageMetadata(t *testing.T) {
+	var distributorCfg Config
+	var clientConfig client.Config
+	var ringConfig ring.Config
+	flagext.DefaultValues(&distributorCfg, &clientConfig, &ringConfig)
+
+	kvStore, closer := consul.NewInMemoryClient(ring.GetCodec(), log.NewNopLogger(), nil)
+	t.Cleanup(func() { assert.NoError(t, closer.Close()) })
+
+	ringConfig.KVStore.Mock = kvStore
+	ingestersRing, err := ring.New(ringConfig, ingester.IngesterRingKey, ingester.IngesterRingKey, log.NewNopLogger(), nil)
+	require.NoError(t, err)
+
+	mock := &mockInstanceClient{}
+	distributorCfg.IngesterClientFactory = ring_client.PoolInstFunc(func(inst ring.InstanceDesc) (ring_client.PoolClient, error) {
+		return mock, nil
+	})
+
+	d, err := New(distributorCfg, clientConfig, validation.MockDefaultOverrides(), nil, ingestersRing, false, nil, log.NewNopLogger())
+	require.NoError(t, err)
+	require.NotNil(t, d)
+
+	ctx := context.Background()
+	ctx = user.InjectOrgID(ctx, "test")
+
+	req := &mimirpb.WriteRequest{
+		Timeseries: []mimirpb.PreallocTimeseries{
+			{TimeSeries: &mimirpb.TimeSeries{
+				Labels:    []mimirpb.LabelAdapter{{Name: model.MetricNameLabel, Value: "test1"}},
+				Exemplars: []mimirpb.Exemplar{},
+			}},
+		},
+		Source: mimirpb.API,
+	}
+
+	err = d.send(ctx, ring.InstanceDesc{Addr: "1.2.3.4:5555", Id: "test"}, req.Timeseries, nil, req.Source)
+	require.NoError(t, err)
+
+	// Verify that d.send added message size to metadata.
+	require.NotEmpty(t, strconv.Itoa(req.Size()), mock.md[grpcutil.MetadataMessageSize])
```

Comment:
I'm not sure this is asserting what we want it to be asserting?

We're asserting that `strconv.Itoa(size)` is not empty but even a request size of `0` will be converted to `"0"` which isn't an empty string. Also I don't understand why we're passing the metadata as the `msgAndArgs` argument to `require.NotEmpty()`.

Reply:
> I'm not sure this is asserting what we want it to be asserting?

You're right, it should be equality test.

> We're asserting that `strconv.Itoa(size)` is not empty but even a request size of `0` will be converted to `"0"` which isn't an empty string. Also I don't understand why we're passing the metadata as the `msgAndArgs` argument to `require.NotEmpty()`.